src/components/PokemonList/index.tsx

```diff
@@ -0,0 +1,88 @@
+import Error from "@components/Error";
+import { ActivityIndicator, StyleSheet } from "react-native";
+import PokemonItem from "../PokemonItem";
+import { ApolloError } from "@apollo/client";
+import { Pokemon } from "../../types";
+import Animated, { FlatListPropsWithLayout } from "react-native-reanimated";
+import SearchBar from "./SearchBar";
+import useScrollHandler from "./hooks/useScrollHandler";
+import NotFound from "./NotFound";
+
+type Props = {
+  pokemons: Pokemon[] | undefined;
+  loading: boolean;
+  error?: ApolloError;
+  fetchMore: () => void;
+  loadingMore: boolean;
+  navigateToPokemon: (name: string) => void;
+  onScroll?: FlatListPropsWithLayout<Pokemon[]>["onScroll"];
+  ListHeaderComponent?: FlatListPropsWithLayout<
+    Pokemon[]
+  >["ListHeaderComponent"];
+  stickyHeaderIndices?: FlatListPropsWithLayout<
+    Pokemon[]
+  >["stickyHeaderIndices"];
+  ListHeaderComponentStyle?: FlatListPropsWithLayout<
+    Pokemon[]
+  >["ListHeaderComponentStyle"];
+};
+
+const PokemonList = ({
+  pokemons,
+  loading,
+  error,
+  fetchMore,
+  loadingMore,
+  navigateToPokemon,
+  onScroll,
+  ListHeaderComponent,
+  stickyHeaderIndices,
+  ListHeaderComponentStyle,
+}: Props) => {
+  if (error) {
+    return <Error />;
+  }
+
+  return (
```

Comment:
Would it be better to have separate component for PokemonList that handle all logic related to the list (including scrolling). And then in All component, you will only use PokemonList and add some additional code for other part of the screen.

Right now, I don't see any benefits from having separate PokemonList component

Reply:
I assume that `PokemonList` is FlatList with a basic configuration, because for each screen we fetch more in the same way, we render the same items, etc. But overall it might be write cleaner